cvelib/cli.py

```diff
@@ -316,21 +355,28 @@ def reject(ctx: click.Context, cve_id: str, cve_json_str: str, print_raw: bool)
     `cnaRejectedContainer` schema in:\n
     https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
     """
-    cve_json = None
-    if cve_json_str:
-        try:
+    if cve_json_file is not None and cve_json_str is not None:
+        raise click.BadParameter(
+            "cannot use both `-f/--cve-json-file` and `-j/--cve-json` to provide a CVE JSON record."
+        )
+
+    try:
+        if cve_json_str is not None:
             cve_json = json.loads(cve_json_str)
-        except json.JSONDecodeError as exc:
-            click.echo("CVE data was not valid JSON. Error was:\n")
-            click.secho(str(exc))
-            return
+        elif cve_json_file is not None:
+            cve_json = json.load(cve_json_file)
+        else:
+            cve_json = None
+    except json.JSONDecodeError as exc:
+        print_error(msg="CVE data is not valid JSON", details=str(exc))
+        return
```

Comment:
This is almost identical to above, so it could be its own "get_data" method, with cve_json_file, cve_json_str, and required=True as arguments. The else case (set cve_json to None for reject, raise error for publish) can be controlled by the required arg. Or you could name it raise_error=True

Reply:
I'll keep this as is since it might be more readable without having to follow the code to its deduplicated version.